### PR TITLE
Enable shifting entries

### PR DESCRIPTION
**Pull request by OmeGak**

### CODE REVIEW

File: indico/modules/events/timetable/util.py [hunk 1]
```diff
@@ -275,24 +275,23 @@ def get_session_block_entries(event, day):
...
-        entries = entries.filter(TimetableEntry.parent_id.is_(None)).all()
+    if entry.parent:
+        query = TimetableEntry.query.with_parent(entry.parent)
+    elif session_:
+        query = event.timetable_entries.filter(TimetableEntry.type == TimetableEntryType.SESSION_BLOCK,
+                                               TimetableEntry.session_block.has(session_id=session_.id))
     else:
-        entries = entries.filter(TimetableEntry.parent == entry.parent).all()
+        query = event.timetable_entries
+    entries = (query.filter(db.cast(TimetableEntry.start_dt.astimezone(tz), db.Date) == entry.start_dt.date(),
```
Comment (ThiefMaster) [inline indico/modules/events/timetable/util.py]: start_dt -> local tz first?

File: indico/modules/events/util.py [hunk 2]
```diff
@@ -385,7 +387,10 @@ def track_time_changes(auto_extend=False):
 
     :param auto_extend: Whether entry parents will get their end_dt
                         automatically extended or not.
+    :param user: The `User` that will trigger time changes.
```
Comment (ThiefMaster) [inline indico/modules/events/util.py]: "The `User` used to perform access checks of cascades operations" or something similar would be more accurate

File: indico/modules/events/util.py [hunk 3]
```diff
@@ -398,9 +403,21 @@ def track_time_changes(auto_extend=False):
     else:
         if auto_extend:
             # g.old_times changes during iteration
+            old = set(g.old_times)
```
Comment (ThiefMaster) [inline indico/modules/events/util.py]: `old` -> `immediate_changes`
`new` -> `cascaded_changes`
?

File: indico/modules/events/util.py [hunk 4]
```diff
@@ -398,9 +403,21 @@ def track_time_changes(auto_extend=False):
     else:
         if auto_extend:
             # g.old_times changes during iteration
```
Comment (ThiefMaster) [inline indico/modules/events/util.py]: this belongs right above the loop

File: indico/modules/events/util.py (hunk 3 above)
Comment (OmeGak) [inline indico/modules/events/util.py]: probably better

File: indico/modules/events/util.py (hunk 2 above)
Comment (OmeGak) [inline indico/modules/events/util.py]: This is a context manager, so the user will be triggering time changes within. I think this is more accurate.

**Pull request merged**